source/grammar/qasm3.g4

```diff
@@ -353,14 +356,17 @@ loopSignature
 
 loopStatement: loopSignature programBlock ;
 
-controlDirectiveStatement
-    : controlDirective SEMICOLON
+endStatement
+    : 'end' SEMICOLON
     ;
 
+returnStatement
+    : 'return' statement;
```

Comment:
I think `statement` should be `statement?` to allow following flow.
```
def sub(int[64] i) {
  if (i < 0) return;
  // do something
}
```

Reply:
Fair pt. Addressed in https://github.com/Qiskit/openqasm/pull/140/commits/9e85ffe8ae570a9769726f6f1a380660beaca65b.